src/ext/php5/serializer.c

```diff
@@ -257,18 +257,79 @@ static void _serialize_exception(zval *el, zval *meta, ddtrace_exception_t *exce
         return;
     }
 
-    int needs_copied = zend_get_object_classname(exception, &class_name, &class_name_len TSRMLS_CC);
-
     add_assoc_long(el, "error", 1);
 
     zend_call_method_with_0_params(&exception, Z_OBJCE_P(exception), NULL, "getmessage", &msg);
-
-    /* add_assoc_stringl does not actually mutate the string, but we've either
-     * already made a copy, or it will when it duplicates with dup param, so
-     * if it did it should still be safe. */
-    add_assoc_stringl(meta, "error.type", (char *)class_name, class_name_len, needs_copied);
     add_assoc_zval(meta, "error.msg", msg);
 
+    bool use_class_name_for_error_type = true;
+    if (instanceof_function(Z_OBJCE_P(exception), ddtrace_ce_fatal_error TSRMLS_CC)) {
+        zval *code = NULL;
+        zend_call_method_with_0_params(&exception, Z_OBJCE_P(exception), NULL, "getcode", &code);
+        if (Z_TYPE_P(code) == IS_LONG) {
```

Comment:
Can `code` still be `NULL` after `zend_call_method_with_0_params`?

Reply:
I'll have to check into this, especially since this is PHP 5.

Reply:
It's not obvious; we had better be safe. Good catch, Sammy.